File: 3.DFS_BFS/Programmers/(43165번)타겟 넘버/[jeunghun] solution.py
answer = 0

def solution(numbers, target):
    def dfs(n, idx, target, numbers):
        global answer

        if idx == len(numbers):
            if n == target:
                answer += 1
            return

        dfs(n - numbers[idx], idx + 1, target, numbers)
        dfs(n + numbers[idx], idx + 1, target, numbers)
    
    dfs(0, 0, target, numbers)
    
    return answer

File: 3.DFS_BFS/Programmers/(43165번)타겟 넘버/[jeunghun] solution.swift
import Foundation

func solution(_ numbers: [Int], _ target:Int) -> Int {
    var answer = 0
    
    func dfs(_ result: Int, idx: Int) {
        if idx == numbers.count {
            if result == target {
                answer += 1
            }
            
            return
        }
        
        dfs(result + numbers[idx], idx: idx + 1)
        dfs(result - numbers[idx], idx: idx + 1)
    }
    
    dfs(0, idx: 0)
    
    return answer
}